src/NightwatchServiceProvider.php

```diff
@@ -357,13 +358,7 @@ private function registerHooks(): void
 
         $events->listen(RequestReceived::class, (new OctaneListener($core))(...)); // @phpstan-ignore class.notFound
 
-        Queue::createPayloadUsing(static fn ($c, $q, array $payload) => [
-            ...$payload,
-            'nightwatch' => [
-                ...($payload['nightwatch'] ?? []),
-                'job_id' => $core->uuid->make(),
-            ],
-        ]);
+        Queue::createPayloadUsing(new CreateQueuePayloadHandler($core));
```

Comment:
I'd prefer to keep this independent of the `PolyfillContextDehydration` hook, rather than mixing them both into a single hook.